packages/gatsby-theme-docs/src/components/multi-code-example.js

```diff
@@ -5,56 +5,72 @@ import {
   MultiCodeBlock,
   CodeBlock,
 } from '@commercetools-docs/ui-kit';
+import CodeExampleLanguageContext from '../hooks/use-code-example-language';
 import useCodeExamples from '../hooks/use-code-examples';
 
 function MultiCodeExample(props) {
   const codeExamples = useCodeExamples();
 
-  try {
-    return (
-      <MultiCodeBlock title={props.title}>
-        {React.Children.map(props.children, (child, index) => {
-          if (!child.props || child.props.mdxType !== 'CodeExample') {
-            throw new Error(
-              `Children of <MultiLanguageCodeExamples> must be a <CodeExample> component and not "${
-                child.props ? child.props.mdxType : child
-              }"`
-            );
-          }
-          if (!child.props.path) {
-            throw new Error(
-              `The component <CodeExample> is missing the required property "path".`
-            );
-          }
-          const codeExample = codeExamples.find((example) => {
-            return example.path === child.props.path;
-          });
-          if (!codeExample) {
-            throw new Error(
-              `The provided path "${child.props.path}" does not exist. Please make sure that the code example file exists at the provided path.`
-            );
-          }
+  return (
+    <CodeExampleLanguageContext.Consumer>
+      {({ language, setLanguage }) => {
+        function handleOnLanguageChange(lang) {
+          setLanguage(lang);
+        }
+        try {
           return (
-            <CodeBlock
-              key={`${codeExample.language}-${index}`}
-              content={codeExample.content}
-              language={codeExample.language}
-              highlightLines={child.props.highlightLines || []}
-              noPromptLines={child.props.noPromptLines || []}
-            />
+            <MultiCodeBlock
+              title={props.title}
+              selectedLanguage={language}
+              onLanguageChange={handleOnLanguageChange}
```

Comment:
I would just move the hook into the `MultiCodeBlock` component, there is no need for this extra wrapping.

Reply:
Please explain how that works.

Reply:
Remove the wrapper with the consumer component, use `React.useContext` within the `MultiCodeBlock` component.

Not sure what else is not clear here.